### ISSUE TITLE
Construction of imaginary time Green's function with np.array of type float gives invalid GF

### ISSUE THREAD

**Issue by chuffa** [TRIQS/triqs#775]
### Description
I have an imaginary time Green's function stored as np.array and would like to calculate the Fourier transform to imaginary frequencies, this fails with a long error list that it cannot find 
a suitable overload:

```TypeError: Error: no suitable C++ overload found in implementation of function gf_factories.make_gf_from_fourier```

As far as I can tell, the imaginary time Green's function is a valid GF.

### Steps to Reproduce
```
beta = 101
data = np.ones((101))
Gtau = GfImTime( mesh = MeshImTime(beta, 'Fermion', len(data)), data = data  )
Giw = make_gf_from_fourier(Gtau)
```
### Versions
You are using the TRIQS library version 2.2.1
You are using the git hash 9b4b09d11e2ca70d446b01c527a3d49eb2c80cb3

**Comment by chuffa** [comment]
The problem lies in the data type of the numpy array, changing it to complex resolves the issue. IMHO the construction of GfImTime should either fail or, even better, the data should be converted to complex.

**Comment by Wentzell** [comment]
This was fixed in bf786019
